### PR TITLE
support go modules

### PR DESCRIPTION
**Pull request by uta-mori**
#13 

Thanks for the great samples.
I have changed it to support go modules.

### CODE REVIEW

**Comment by Chuxel** [comment]
Thanks for the contribution!  I think it would be best to leave the Dockerfile in such a state that it can be used without modules if appropriate. The Go tools are also installed in an alternate location and the source removed afterward to avoid conflicts with the actual Go application. These "try" samples are often used by customers to start their own projects as well, so we tend to leave them like what is in https://github.com/microsoft/vscode-dev-containers

@hyangah What do you think about the other updates?  Moving the sample to modules makes sense to me.

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -1,31 +1,52 @@
...
-		// Update the VARIANT arg to pick a version of Go
-		"args": { "VARIANT": "1" }
-	},
-	"runArgs": [ "--cap-add=SYS_PTRACE", "--security-opt", "seccomp=unconfined" ],
+  "name": "Go",
+  "build": {
+    "dockerfile": "Dockerfile",
+    // Update the VARIANT arg to pick a version of Go
+    "args": { "VARIANT": "1.14" }
+  },
```
Comment (hyangah) [inline .devcontainer/devcontainer.json:7]: Leave this as it is - 1.15 release is pretty close.
I think setting 1 will pick up the container with the latest stable go.

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -1,31 +1,52 @@
...
+			// the completion engine is allowed to make suggestions for packages that you do not currently import.
+			"completeUnimported": true,
 
-	// Use 'forwardPorts' to make a list of ports inside the container available locally.
-	"forwardPorts": [9000],
+      // If true, enable additional analyses with staticcheck.
+      // Warning: This will significantly increase memory usage.
+      "staticcheck": true
+    },
+    "go.lintTool": "golangci-lint"
```
Comment (hyangah) [inline .devcontainer/devcontainer.json:38]: Just go with the default lint tool.

**Comment by uta-mori** [comment]
> Thanks for the contribution! I think it would be best to leave the Dockerfile in such a state that it can be used without modules if appropriate. The Go tools are also installed in an alternate location and the source removed afterward to avoid conflicts with the actual Go application. These "try" samples are often used by customers to start their own projects as well, so we tend to leave them like what is in https://github.com/microsoft/vscode-dev-containers
> 
> @hyangah What do you think about the other updates? Moving the sample to modules makes sense to me.

thanks.
I've seen sample projects in other languages.
There, dependency files were also included in the project. (`Cargo.toml`,`package.json`)

and
Packages that don't support modules may not need to be installed.

 https://github.com/golang/vscode-go/blob/master/docs/tools.md#formatting

> NOTE: goreturns does not have support for Go modules, so we recommend using goimports or gopls instead.


At any rate, I learned a lot. Thank you.

**Comment by Chuxel** [comment]
> thanks.
> I've seen sample projects in other languages.
> There, dependency files were also included in the project. (Cargo.toml,package.json)

Yeah, the vscode-dev-containers repo populate an in product experience that filters out everything but the `.devcontainer` folder. Manual instructions tell you to just include the `.devcontainer` folder as well.  So it's those contents that end up here.   So let's leave those as is as hyangah suggested.

File: .devcontainer/devcontainer.json (hunk 2 above)
Comment (uta-mori) [reply .devcontainer/devcontainer.json:38]: ok

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (uta-mori) [reply .devcontainer/devcontainer.json:7]: Yes, you're right.

File: .vscode/settings.json [hunk 3]
```diff
@@ -1,4 +1,4 @@
 {
     "go.useLanguageServer": true,
-    "go.inferGopath": true
-}
\ No newline at end of file
+    "go.inferGopath": false
```
[the remark this answers is not in the document]
Comment (uta-mori) [reply .vscode/settings.json:3]: I didn't have to.

**Comment by Chuxel** [comment]
LGTM!

**Pull request merged**